pkg/cmd/roachtest/test_runner.go

```diff
@@ -1498,7 +1498,7 @@ func (r *testRunner) postTestAssertions(
 			}
 
 			if db == nil {
-				db = c.Conn(ctx, t.L(), s.Node)
```

Comment:
@stevendanna You were saying we need two `Conn`s here? A "system" one for the consistency check, and whatever is the other one for the `CheckInvalidDescriptors`?

Reply:
`CheckInvalidDescriptors` also fetches from a `crdb_internal` table, so I'm not sure. Trying a common "system" connection for now.

Reply:
Yeah, the check invalid descriptor check will _work_ against the system tenant, but likely isn't testing as much as we want. That function works by reading the system.descriptor table and running all sorts of validations.  That table is a per-tenant table. So the system tenant and it is good to validate. But we probably also want to validate whatever the main application tenant is.

Reply:
I'm not sure how the system/non-system plumbing works here and how to avoid doing the same work twice in `SystemOnlyDeployment`. Using the default conn for `CheckInvalidDescriptors` for now (will it use the non-system tenant automatically?), and can add a system one later?

Reply:
> sing the default conn for CheckInvalidDescriptors for now (will it use the non-system tenant automatically?), and can add a system one later?

That works for me.  I think it'll do the right thing for a lot of tests.